packages/flutter/lib/src/material/input_border.dart

```diff
@@ -246,9 +246,45 @@ class UnderlineInputBorder extends InputBorder {
     TextDirection? textDirection,
   }) {
     if (borderRadius.bottomLeft != Radius.zero || borderRadius.bottomRight != Radius.zero) {
-      canvas.clipPath(getOuterPath(rect, textDirection: textDirection));
+      _drawMultipleWidth(canvas, borderRadius.toRRect(rect), borderSide);
+    } else {
+      canvas.drawLine(rect.bottomLeft, rect.bottomRight, borderSide.toPaint());
     }
-    canvas.drawLine(rect.bottomLeft, rect.bottomRight, borderSide.toPaint());
+  }
+
+  void _drawMultipleWidth(Canvas canvas, RRect borderRect, BorderSide bottom) {
+    final Paint paint = Paint()..color = bottom.color;
+    final RRect inner = _inflateRRect(
+        borderRect, EdgeInsets.fromLTRB(0, 0, 0, bottom.strokeInset));
+    final RRect outer = _deflateRRect(
+        borderRect, EdgeInsets.fromLTRB(0, 0, 0, bottom.strokeOutset));
+    canvas.drawDRRect(outer, inner, paint);
+  }
+
+  static RRect _inflateRRect(RRect rect, EdgeInsets insets) {
+    return RRect.fromLTRBAndCorners(
+      rect.left - insets.left,
+      rect.top - insets.top,
+      rect.right + insets.right,
+      rect.bottom + insets.bottom,
+      topLeft: (rect.tlRadius + Radius.elliptical(insets.left, insets.top)).clamp(minimum: Radius.zero), // ignore_clamp_double_lint
```

Comment:
Why not use [clampDouble](https://api.flutter.dev/flutter/package-material_color_utilities_utils_math_utils/MathUtils/clampDouble.html) on these?

Reply:
Because there are two values, so the other possible way would be

Radius(clampDouble(min: 0, rect.tlRadiusX + insets.left), clampDouble(min: 0, rect.tlRadiusY + inset.top))

Do you prefer that way? I can make it.

Reply:
Oh, good news: I copied the code from what I'm doing on BoxDecoration. When my other PR gets merged, this will be redundant and I'll be able to use the `BoxDecoration.paintNonUniformBorder`.

Reply:
I guess the clampDouble way doesn't look much worse, I say use that.  The linter rule is there because `.clamp` is much slower, see https://github.com/flutter/flutter/issues/103917.